# **Campus Recruitment - Academic and Employability Factors influencing placement**

# ***A model to predict if a student gets placed or not***

In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import seaborn as sns

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 5GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/factors-affecting-campus-placement/Placement_Data_Full_Class.csv


***Load the dataset***

In [2]:
data=pd.read_csv("../input/factors-affecting-campus-placement/Placement_Data_Full_Class.csv",index_col='sl_no')

In [3]:
data.head()

gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p   degree_t  \
sl_no                                                                         
1          M  67.00   Others  91.00   Others  Commerce     58.00   Sci&Tech   
2          M  79.33  Central  78.33   Others   Science     77.48   Sci&Tech   
3          M  65.00  Central  68.00  Central      Arts     64.00  Comm&Mgmt   
4          M  56.00  Central  52.00  Central   Science     52.00   Sci&Tech   
5          M  85.80  Central  73.60  Central  Commerce     73.30  Comm&Mgmt   

      workex  etest_p specialisation  mba_p      status    salary  
sl_no                                                              
1         No     55.0         Mkt&HR  58.80      Placed  270000.0  
2        Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
3         No     75.0        Mkt&Fin  57.80      Placed  250000.0  
4         No     66.0         Mkt&HR  59.43  Not Placed       NaN  
5         No     96.8        Mkt&Fin  55.50      Placed  425000.0

In [4]:
x=data.iloc[:,:12].values
y=data.iloc[:,-2].values

***Let's now explore the features***

**Categorical Features:**
* Gender
* ssc_b
* hsc_b
* hsc_s
* degree_t
* workex
* specialisation
* status

**Numerical Features:**
* ssc_p
* hsc_p
* degree_p
* etest_p
* mba_p
* salary

***Let's check for missing values***

In [5]:
data.isnull().sum()

gender             0
ssc_p              0
ssc_b              0
hsc_p              0
hsc_b              0
hsc_s              0
degree_p           0
degree_t           0
workex             0
etest_p            0
specialisation     0
mba_p              0
status             0
salary            67
dtype: int64

***Encode the categorical features using OneHotEncoding***

In [6]:
data['ssc_b'].unique()

array(['Others', 'Central'], dtype=object)

In [7]:
data['hsc_b'].unique()

array(['Others', 'Central'], dtype=object)

In [8]:
data['hsc_s'].unique()

array(['Commerce', 'Science', 'Arts'], dtype=object)

In [9]:
data['degree_t'].unique()

array(['Sci&Tech', 'Comm&Mgmt', 'Others'], dtype=object)

In [10]:
data['specialisation'].unique()

array(['Mkt&HR', 'Mkt&Fin'], dtype=object)

In [11]:
from sklearn.preprocessing import OneHotEncoder
encoder=OneHotEncoder()
x=encoder.fit_transform(x)

In [12]:
from sklearn.preprocessing import LabelEncoder
enc=LabelEncoder()
y=enc.fit_transform(y)

Split the training data to test and train so as to find the best algorithm based on accuracy.

In [13]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,random_state=20,test_size=0.2)

***Random Forest Classifier***

In [14]:
from sklearn.ensemble import RandomForestClassifier
rf=RandomForestClassifier(n_estimators=10, max_depth=5, random_state=1)

rf.fit(x_train,y_train)
accuracy=rf.score(x_test,y_test)

print("Random Forest accuracy is :{}".format(accuracy))

Random Forest accuracy is :0.627906976744186


***Naive Bayes Classifier***

In [15]:
from sklearn.naive_bayes import MultinomialNB
# Instantiate the classifier
mnb = MultinomialNB()

# Train classifier
mnb.fit( x_train,y_train)
accuracy=mnb.score(x_test,y_test)
print("Naive bayes accuracy is :{}".format(accuracy))

Naive bayes accuracy is :0.8372093023255814


***K-Nearest Neighbors Classifier***

In [16]:
from sklearn.neighbors import KNeighborsClassifier
knn=KNeighborsClassifier(n_neighbors=29)            #n_neighbors optimal value should be suqare root of n
knn.fit(x_train,y_train)
y_pred_knn=knn.predict(x_test)

#finding accuracy and confusion matrix
accuracy=knn.score(x_test,y_test)
print("KNN accuracy is :{}".format(accuracy))

KNN accuracy is :0.7209302325581395


***Suppor Vectore Machine with different kernels***

In [17]:
from sklearn import svm    			
C = 0.6  # SVM regularization parameter
svc = svm.SVC(kernel='linear', C=C).fit(x_train, y_train)
#svc = svm.LinearSVC(C=C).fit(X, y)
rbf_svc = svm.SVC(kernel='rbf', gamma=0.5, C=C).fit(x_train, y_train)
# SVC with polynomial (degree 3) kernel
poly_svc = svm.SVC(kernel='poly', degree=2, C=C).fit(x_train, y_train)
accuracy1=svc.score(x_test,y_test)
print("SVM accuracy is :{}".format(accuracy1))
accuracy2=rbf_svc.score(x_test,y_test)
print("SVM rbf accuracy is :{}".format(accuracy2))
accuracy3=poly_svc.score(x_test,y_test)
print("SVM poly accuracy is :{}".format(accuracy3))

SVM accuracy is :0.7674418604651163
SVM rbf accuracy is :0.6976744186046512
SVM poly accuracy is :0.6976744186046512


***Decision Tree Classifier***

In [18]:
from sklearn.tree import DecisionTreeClassifier
dtree = DecisionTreeClassifier(max_depth = 6,random_state = 99, max_features = None, min_samples_leaf = 5)
dtree.fit(x_train,y_train)
accuracy=dtree.score(x_test,y_test)
print("Decision tree accuracy is :{}".format(accuracy))

Decision tree accuracy is :0.6744186046511628


***Logistic Regression***

In [19]:
from sklearn.linear_model import LogisticRegression
lr=LogisticRegression()
lr.fit(x_train,y_train)
accuracy=lr.score(x_test,y_test)
print("Decision tree accuracy is :{}".format(accuracy))


Decision tree accuracy is :0.7674418604651163


***Clearly, Naive Bayes Clasifier gives the highest accuracy of 83.72%***

Fit the entire dataset with the Naive Bayes Classifier

In [20]:
from sklearn.naive_bayes import MultinomialNB
model = MultinomialNB()
model.fit(x,y)


MultinomialNB()